File: files/cgi-bin/getTime.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-

import datetime
import time

# pour tester le timeout
time.sleep(15)

# En-tête HTTP obligatoire pour CGI
print("Content-Type: text/html; charset=utf-8")
print()  # Ligne vide pour séparer les en-têtes du contenu

# Génère la page HTML avec l'heure actuelle
print("<html>")
print("<head><title>CGI Test</title></head>")
print("<body>")
print("<h1>CGI Script Execution Successful</h1>")
print(f"<p>The current server time is: <b>{datetime.datetime.now()}</b></p>")
print("</body>")
print("</html>")

File: files/cgi-bin/getTime.c
#include <stdio.h>
#include <time.h>

int main(void) {
    // 1. Get the current system time
    time_t now = time(NULL);
    char *time_str = ctime(&now);

    // 2. Print HTTP Headers (Required for CGI)
    // No spaces between header name and colon
    printf("Content-Type: text/html; charset=utf-8\r\n");
    printf("Status: 200 OK\r\n");
    
    // 3. The empty line that separates headers from the body
    printf("\r\n");

    // 4. The HTML Body
    printf("<!DOCTYPE html>\n");
    printf("<html>\n<head><title>CGI Time</title></head>\n");
    printf("<body>\n");
    printf("<h1>Current Server Time</h1>\n");
    printf("<p>The time is: <strong>%s</strong></p>\n", time_str);
    printf("</body>\n</html>\n");

    return 0;
}